nslsii/EPS_devices/valves_and_shutters.py

```diff
@@ -0,0 +1,339 @@
+                    cmd_sig.set(1)
+                    ts = datetime.datetime.fromtimestamp(
+                        timestamp).strftime(self._time_fmtstr)
+                    print(f'** ({ts}) Had to reactuate {self.name} while'
+                          f'moving to {val}')
+                else:
+                    cmd_sig.clear_sub(cmd_retry_cb)
+
+        cmd_sig.subscribe(cmd_retry_cb, run=False)
+        cmd_sig.set(1)
+        self.status.subscribe(state_cb)
+
+        return st
+
+    def stop():
+        self.status.clear_sub(state_cb)
+        self._set_st._finished(success=False)
+        super().stop()
+
+class PneumaticActuator(EPSTwoStateDevice):
+    '''An ``ophyd.Device`` for EPS driven pneumatic actuators at NSLS-II.
+
+
+    This is for use with NSLS-II EPS based pneumatic actuators that have two
+    distinct states 'In' and 'Out' and are actuated into these two states by
+    the PV's ``'{prefix}Cmd:In-Cmd'`` and ``'{prefix}Cmd:out-Cmd'``.
+
+    The basic use case is to define the device using the line:
+
+    ``my_pneumatic_actuator = PneumaticActuator(prefix='a_PV_prefix',
+                                                name='my_pneumatic_actuator')``
+
+    which can then be actuated by:
+    ``RE(mv(my_pneumatic_actuator, 'in')`` or
+    ``RE(mv(my_pneumatic_actuator, 'out')``
+
+    Parameters
+    ----------
+    prefix : str, keyword only
+        The PV prefix for all components of the device
+    name : str, keyword only
+        The name of the device as will be reported via read()
+    state1_val : float, int or str, optional
+        The value displayed when the device is read that signifys state 1
+    state2_val : float, int or str, optional
+        The value displayed when the device is read that signifys state 2
+    state1_str : str, optional
+        The string value to be passed to ``RE(mv(device, val))`` to move to
+        state 1
+    state2_str : str, optional
+        The string value to be passed to ``RE(mv(device, val))`` to move to
+        state 2
+    num_retries : int, optional
+        This number of attempts at changing the state prior to raising an
+        error, the default is 1.
+    retry_sleep_time : float, optional
+        This is the time in seconds to wait between retries on changing the
+        state, the default is 0.5 s.
+    stop_str : str or None, optional
+        The value of ``state1_str`` or ``state2_str`` to be called in the
+        ``stop`` method. The default, if None is entered, is ``state2_str``.
+    **kwargs : dict, optional
+        The kwargs passed to the ``ophyd.Device`` ``__init__`` method
+    '''
+
+    def __init__(self, *, prefix, name,
+                 state1_val='In', state2_val='Out',
+                 state1_str='in', state2_str='out',
+                 num_retries=1, retry_sleep_time=0.5,
+                 stop_str=None, **kwargs):
+
+        kwargs = dict(state1_pv_uid='In', state2_pv_uid='Out',
```

Comment:
~~Why can't we use the `state1_val` and `state2_val` instead of `In` and `Out` respectively?~~
I saw another use like this in https://github.com/NSLS-II/nslsii/pull/52/files#diff-e886eb2e4c555c0292a7ed27bc136211R275, disregard my comment.